kartothek/core/dataset.py

```diff
@@ -888,6 +888,28 @@ def from_dataset(dataset):
         ds_builder.tables = dataset.tables
         return ds_builder
 
+    def modify_dataset_name(self, new_ds_name):
+        """
+        Modify the dataset name and change dependent fields, i.e. UUID and partitioning
+        information.
+
+        TODO: This will atm only work for cube datasets with a UUID like cube++dataset.
+        Which UUIDs are possible for non-cube datasets?
+        """
+        rx_transform_path = re.compile(r"(\w*)\+\+(\w*)/(.*)")
```

Comment:
this class should not make *any* assumptions about the UUID. Everything that is allowed by https://github.com/JDASoftwareGroup/kartothek/blob/8848fd43639ea10785a7fe87833a7b6dc687850e/kartothek/core/dataset.py#L36-L37 can be a UUID and this class should handle it this way

Reply:
Not accounting for performance at this moment, you can use https://github.com/JDASoftwareGroup/kartothek/blob/8848fd43639ea10785a7fe87833a7b6dc687850e/kartothek/core/urlencode.py#L25-L39 to split a given key into it's components